tests/org.jboss.tools.deltaspike.ui.bot.test/src/org/jboss/tools/deltaspike/ui/bot/test/DeltaspikeTestBase.java

```diff
@@ -75,45 +79,68 @@ public static void disableSourceLookup(){
 		preferenceDialog.ok();
 	}
 
-	protected static void importDeltaspikeProject(String projectName,ServerRequirement sr) {
-		
+	/**
+	 * Import project, configure project to use given target platform and check
+	 * if CDI support is enabled. If isn't, then enable CDI support and
+	 * revalidate project.
+	 * 
+	 * @param projectName
+	 *            Project to import
+	 * @param sr
+	 *            The target platform which should be used
+	 */
+	protected static void importDeltaspikeProject(String projectName, ServerRequirement sr) {
+
 		try {
 			importMavenProject(projectName);
 		} catch (IOException e) {
 			e.printStackTrace();
-			fail("Unable to import "+projectName);
+			fail("Unable to import " + projectName);
 		}
-		
+
 		ProjectExplorer pe = new ProjectExplorer();
 		pe.open();
-		pe.getProject(projectName).select();
-		new ContextMenu("Properties").select();
-		new DefaultShell("Properties for "+projectName);
-		new DefaultTreeItem("Targeted Runtimes").select();
-		for(TableItem i: new DefaultTable().getItems()){
-			if(i.getText().equals(sr.getRuntimeNameLabelText(sr.getConfig()))){
-				i.setChecked(true);
-			} else {
-				i.setChecked(false);
-			}
+		Project explorerItem = pe.getProject(projectName);
+
+		PropertyDialog dialog = new ExplorerItemPropertyDialog(explorerItem);
+		RuntimesPropertyPage propertyPage = new RuntimesPropertyPage();
+		String targetRuntime = sr.getRuntimeNameLabelText(sr.getConfig());
+
+		dialog.open();
+		dialog.select(propertyPage);
+		propertyPage.selectRuntime(targetRuntime);
+
+		new DefaultTreeItem("CDI (Contexts and Dependency Injection) Settings").select();
```

Comment:
Im still missing "CDI (Contexts and Dependency Injection) Settings" PropertyPage :)